util/fipstools/acvp/acvptool/subprocess/pbkdf.go

```diff
@@ -0,0 +1,141 @@
+var iterationMax uint64 = 10000000
+var saltLenMin int = 128
+var saltLenMax int = 4096
+var passwordLenMin int = 8
+var passwordLenMax int = 4096
+
+// List of supported PBKDF test types by NIST
+var TestTypeList map[string]bool = map[string]bool{
+	"AFT": true,
+}
+
+func (a *pbkdf) Process(vectorSet []byte, m Transactable) (interface{}, error) {
+	var parsed pbkdfVectorSet
+	if err := json.Unmarshal(vectorSet, &parsed); err != nil {
+		return nil, err
+	}
+
+	var ret []pbkdfTestGroupResponse
+	// See draft-celi-acvp-symmetric.html#table-6. (NIST no longer publish HTML
+	// versions of the ACVP documents. You can find fragments in
+	// https://github.com/usnistgov/ACVP.)
+	for _, group := range parsed.Groups {
+		response := pbkdfTestGroupResponse{
+			ID: group.ID,
+		}
+
+		// Check for valid HMAC Algorithm
+		_, algIn := HmacAlgList[group.HmacAlg]
+		if !algIn {
+			return nil, fmt.Errorf("test group %d has unsupported HMAC algorithm %q", group.ID, group.HmacAlg)
+		}
+
+		// Check if the test type is valid
+		_, typeIn := TestTypeList[group.Type]
+		if !typeIn {
+			return nil, fmt.Errorf("test group %d has unsupported test type %q", group.ID, group.Type)
+		}
+
+		// Check each test in the test group
+		for _, test := range group.Tests {
+
+			if test.KeyLen > keyLenMax || test.KeyLen < keyLenMin {
+				return nil, fmt.Errorf("test case %d/%d requests output key of length %d, but expected output key length between %d and %d", group.ID, test.ID, test.KeyLen, keyLenMin, keyLenMax)
+			}
+
+			// check length of salt against NIST min/max
+			if len(test.SaltHex)*4 > saltLenMax || len(test.SaltHex)*4 < saltLenMin {
+				return nil, fmt.Errorf("test case %d/%d contains salt %q of bit length %d, but expected bit length between %d and %d", group.ID, test.ID, test.SaltHex, len(test.SaltHex)*4, saltLenMin, saltLenMax)
+			}
+
+			if len(test.Pwd) > passwordLenMax || len(test.Pwd) < passwordLenMin {
+				return nil, fmt.Errorf("test case %d/%d contains password %s of length %d, but expected password length between %d and %d", group.ID, test.ID, test.Pwd, len(test.Pwd), passwordLenMin, passwordLenMax)
+			}
+
+			if test.Iterations > iterationMax || test.Iterations < iterationMin {
+				return nil, fmt.Errorf("test case %d/%d contains iteration count of %d, but expected iteration count between %d and %d", group.ID, test.ID, test.Iterations, iterationMin, iterationMax)
+			}
+
+			testResp := pbkdfTestResponse{ID: test.ID}
+
+			// encode hex
+			salt, salt_err := hex.DecodeString(test.SaltHex)
+			if salt_err != nil {
+				return nil, fmt.Errorf("Salt decode failed: %s", salt_err)
+			}
+
+			var result [][]uint8
+			result, err := m.Transact("PBKDF", 1, []byte(test.Pwd),
+				salt, uint32le(uint32(test.Iterations)), []byte(group.HmacAlg),
+				uint32le(uint32(test.KeyLen)))
+			if err != nil {
+				return nil, fmt.Errorf("PBKDF operation failed: %s", err)
+			}
+
+			// debug putting all letterse to upper in the output for now
```

Comment:
Not sure I get what you mean here?

Reply:
Was an old debug statement that slipped through the cracks, removed :)